### PR TITLE
Remove documentation for XMLSec1 installation

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: https://github.com/mattermost/docs/issues/4359

Updated:
- Self-Managed Admin Guide > Configure Mattermost > Configuration Settings > Use New SAML Library
   - Marked this configuration setting as deprecated as of December 2020

### CODE REVIEW

File: source/deployment/sso-saml-adfs-msws2016.rst [hunk 1]
```diff
@@ -10,79 +10,79 @@ This document provides steps to configure SAML 2.0 with Microsoft ADFS for Matte
...
 
 	.. image:: ../../source/images/SSO-SAML-ADFS_add-new-relying-party-trust_004.png
 
-If you would like to set up encryption for your SAML connection, select **Browse** and upload your Service Provider Public Certificate.
+If you would like to set up encryption for your SAML connection, select **Browse**, then upload your Service Provider Public Certificate.
 
 	.. image:: ../../source/images/SSO-SAML-ADFS_add-new-relying-party-trust_005.png
 
-6. In the **Configure URL** screen, select **Enable Support for the SAML 2.0 WebSSO protocol** and enter the **SAML 2.0 SSO service URL** in the following format:``https://<your-mattermost-url>/login/sso/saml`` where ``<your-mattermost-url>`` should typically match the `Mattermost Site URL <https://docs.mattermost.com/administration/config-settings.html#site-url>`__.
+6. On the **Configure URL** screen, select **Enable Support for the SAML 2.0 WebSSO protocol**, then enter the **SAML 2.0 SSO service URL** in the following format:``https://<your-mattermost-url>/login/sso/saml`` where *<your-mattermost-url>* should typically match the `Mattermost Site URL <https://docs.mattermost.com/administration/config-settings.html#site-url>`__.
```
Comment (justinegeffen) [inline source/deployment/sso-saml-adfs-msws2016.rst:48]:
```suggestion
6. On the **Configure URL** screen, select **Enable Support for the SAML 2.0 WebSSO protocol**, then enter the **SAML 2.0 SSO service URL** in the following format:``https://<your-mattermost-url>/login/sso/saml`` where ``<your-mattermost-url>`` should typically match the `Mattermost Site URL <https://docs.mattermost.com/administration/config-settings.html#site-url>`__.
```

File: source/deployment/sso-saml-adfs-msws2016.rst [hunk 2]
```diff
@@ -93,63 +93,63 @@ From Mattermost version 3.4, the ``FirstName`` and ``LastName`` attributes are o
...
-7. Click **Finish** to create the claim rule, then **OK** to finish creating rules.
+7. Select **Finish** to create the claim rule, then select **OK** to finish creating rules.
 
-8. Open Windows PowerShell as an administrator and run the following command:
+8. Open Windows PowerShell as an administrator, then run the following command:
 
   ``Set-ADFSRelyingPartyTrust -TargetName <display-name> -SamlResponseSignature "MessageAndAssertion"``
 
-where *<display-name>* is the name you specified in step 4 of *Add a Relying Party Trust*. In our example it would be ``mattermost``.
+where *<display-name>* is the name you specified in step 4 when you added a relying party trust*. In this example,  *<display-name>* would be ``mattermost``.
```
Comment (justinegeffen) [inline source/deployment/sso-saml-adfs-msws2016.rst:121]:
```suggestion
where ``<display-name>``is the name you specified in step 4 when you added a relying party trust. In this example, ``<display-name>`` would be ``mattermost``.
```

File: source/deployment/sso-saml-adfs-msws2016.rst [hunk 3]
```diff
@@ -194,15 +194,15 @@ For Mattermost servers running 3.3 and earlier, the first name and last name att
 
 	.. image:: ../../source/images/SSO-SAML-ADFS_configure-saml_004.png
 
-5. Click **Save**.
+5. Select **Save**.
 
-6. (Optional) If you configured First Name Attribute and Last Name Attribute, go to **System Console > Site Configuration > Users and Teams** (or **System Console > General > Users and Teams** in versions prior to 5.12) and set **Teammate Name Display** to **Show first and last name**. This is recommended for a better user experience.
+6. (Optional) If you configured ``First Name`` Attribute and the ``Last Name`` Attribute, go to **System Console > Site Configuration > Users and Teams** (or **System Console > General > Users and Teams** in versions prior to 5.12), then set **Teammate Name Display** to **Show first and last name**. This is recommended for a better user experience.
```
Comment (justinegeffen) [inline source/deployment/sso-saml-adfs-msws2016.rst:199]:
```suggestion
6. (Optional) If you configured ``First Name`` Attribute and the ``Last Name`` Attribute, go to **System Console > Site Configuration > Users and Teams**, then set **Teammate Name Display** to **Show first and last name**. This is recommended for a better user experience.
```

File: source/deployment/sso-saml-adfs.rst [hunk 4]
```diff
@@ -92,61 +92,61 @@ Create Claim Rules
...
-7. Click **Finish** to create the claim rule, then **OK** to finish creating rules.
+7. Select **Finish** to create the claim rule, then select **OK** to finish creating rules.
 
-8. Open Windows PowerShell as an administrator and run the following command:
+8. Open Windows PowerShell as an administrator, then run the following command:
 
   ``Set-ADFSRelyingPartyTrust -TargetName <display-name> -SamlResponseSignature "MessageAndAssertion"``
 
-where <display-name> is the name you specified in step 7. In our example it would be ``mattermost``.
+where *<display-name>* is the name you specified in step 4 when adding a relying party trust. In this example, *<display-name>* would be ``mattermost``.
```
Comment (justinegeffen) [inline source/deployment/sso-saml-adfs.rst:122]:
```suggestion
where ``<display-name>`` is the name you specified in step 4 when adding a relying party trust. In this example, ``<display-name>`` would be ``mattermost``.
```

File: source/deployment/sso-saml-adfs.rst [hunk 5]
```diff
@@ -178,23 +178,23 @@ Alternatively you can enter the following fields manually:
...
 
 6. (Optional) Customize the login button text.
 
   .. image:: ../../source/images/adfs_26_mattermost_login_button.PNG
 
-7. Click **Save**.
+7. Select **Save**.
 
-8. (Optional) If you configured First Name Attribute and Last Name Attribute, go to **System Console > Site Configuration > Users and Teams** (or **System Console > General > Users and Teams** in versions prior to 5.12) and set **Teammate Name Display** to **Show first and last name**. This is recommended for a better user experience.
+8. (Optional) If you configured a ``FirstName`` and ``LastName`` Attribute, go to **System Console > Site Configuration > Users and Teams** (or **System Console > General > Users and Teams** in versions prior to 5.12), then set **Teammate Name Display** to **Show first and last name**. This is recommended for a better user experience.
```
Comment (justinegeffen) [inline source/deployment/sso-saml-adfs.rst:191]:
```suggestion
8. (Optional) If you configured a ``FirstName`` and ``LastName`` Attribute, go to **System Console > Site Configuration > Users and Teams**, then set **Teammate Name Display** to **Show first and last name**. This is recommended for a better user experience.
```

File: source/deployment/sso-saml-before-you-begin.rst [hunk 6]
```diff
@@ -1,14 +1,7 @@
...
-1. Make sure you have the `XML Security Library <https://www.aleksey.com/xmlsec/download.html>`__ installed on your Mattermost instance. The XML Security Library is usually included as part of Debian GNU/Linux.
-
-2. Install the *xmlsec1-openssl* library
- - On Ubuntu: ``sudo apt-get install xmlsec1``
- - On RHEL: ``sudo yum install xmlsec1-openssl``
-
-3. Generate encryption certificates for encrypting the SAML connection.
-  a. You can use the `Bash script <https://github.com/mattermost/docs/tree/master/source/scripts/generate-certificates>`__ from the *mattermost/docs* repository on GitHub, or any other suitable method.
-  b. Save the two files that are generated. They are the private key and the public key. In the System Console, they are referred to as the **Service Provider Private Key** and the **Service Provider Public Certificate** respectively.
+1. You can use the `Bash script <https://github.com/mattermost/docs/tree/master/source/scripts/generate-certificates>`__ from the *mattermost/docs* repository on GitHub, or any other suitable method.
```
Comment (justinegeffen) [inline source/deployment/sso-saml-before-you-begin.rst:6]:
```suggestion
1. You can use the `Bash script <https://github.com/mattermost/docs/tree/master/source/scripts/generate-certificates>`__ from the ``mattermost/docs`` repository on GitHub, or any other suitable method.
```

File: source/deployment/sso-saml-okta.rst [hunk 7]
```diff
@@ -97,15 +98,15 @@ For Mattermost servers running 3.3 and earlier, the first name and last name att
 
 	.. image:: ../../source/images/okta_14_mattermost_login_button.PNG
 
-7. Click **Save**.
+7. Select **Save**.
 
-8. (Optional) If you configured First Name Attribute and Last Name Attribute, go to **System Console** > **Site Configuration** > **Users and Teams** (or **System Console > General > Users and Teams** in versions prior to 5.12) and set **Teammate Name Display** to *Show first and last name*. This is recommended for a better user experience.
+8. (Optional) If you configured ``First Name`` Attribute and ``Last Name`` Attribute, go to **System Console > Site Configuration > Users and Teams** (or **System Console > General > Users and Teams** in versions prior to 5.12), then set **Teammate Name Display** to **Show first and last name**. This is recommended for a better user experience.
```
Comment (justinegeffen) [inline source/deployment/sso-saml-okta.rst:103]:
```suggestion
8. (Optional) If you configured ``First Name`` Attribute and ``Last Name`` Attribute, go to **System Console > Site Configuration > Users and Teams**, then set **Teammate Name Display** to **Show first and last name**. This is recommended for a better user experience.
```

File: source/deployment/sso-saml-onelogin.rst [hunk 8]
```diff
@@ -111,21 +111,21 @@ Configure SAML sign-in for Mattermost
...
 
   .. image:: ../../source/images/okta_13_mattermost_attributes.PNG
 
 6. (Optional) Customize the login button text.
 
-7. Click **Save**.
+7. Select **Save**.
 
-8. (Optional) If you configured First Name Attribute and Last Name Attribute, go to **System Console > Site Configuration > Users and Teams** (or **System Console > General > Users and Teams** in versions prior to 5.12) and set **Teammate Name Display** to *Show first and last name*. This is recommended for a better user experience.
+8. (Optional) If you configured ``First Name`` Attribute and ``Last Name`` Attribute, go to **System Console > Site Configuration > Users and Teams** (or **System Console > General > Users and Teams** in versions prior to 5.12), then set **Teammate Name Display** to **Show first and last name**. This is recommended for a better user experience.
```
Comment (justinegeffen) [inline source/deployment/sso-saml-onelogin.rst:122]:
```suggestion
8. (Optional) If you configured ``First Name`` Attribute and ``Last Name`` Attribute, go to **System Console > Site Configuration > Users and Teams**, then set **Teammate Name Display** to **Show first and last name**. This is recommended for a better user experience.
```

File: source/deployment/sso-saml-adfs-msws2016.rst [hunk 9]
```diff
@@ -158,16 +158,16 @@ Configure SAML Sign-in for Mattermost
...
 
-Next, start Mattermost server and sign into Mattermost as a System Administrator. Go to **System Console > Authentication > SAML**, paste the  metadata URL in the **Identity Provider Metadata URL** field, and then select **Get SAML Metadata from IdP**.
+Next, start the Mattermost server, then sign into Mattermost as a System Administrator. Go to **System Console > Authentication > SAML**, paste the metadata URL in the **Identity Provider Metadata URL** field, then select **Get SAML Metadata from IdP**.
 
-This populates the **SAML SSO URL** and the **Identity Provider Issuer URL** fields automatically and the Identity Provider Public Certificate is also downloaded from the server and set locally.
+This populates the **SAML SSO URL** and the **Identity Provider Issuer URL** fields automatically. The Identity Provider Public Certificate is also downloaded from the server and set locally.
 
 The following fields can be selected:
   - Set **Enable Login With SAML 2.0** to ``true``.
```
Comment (justinegeffen) [inline source/deployment/sso-saml-adfs-msws2016.rst:166]:
```suggestion
  - Set **Enable Login With SAML 2.0** to **true**.
```

**Review by justinegeffen — COMMENTED** [review]
Great PR, @cwarnermm! As per our discussion around the 5.12 caveat, I've gone ahead and removed instances of them from this PR. Other comments are primarily formatting - the content is great. :)

**Review by justinegeffen — APPROVED** [review]

**Review by adamjclarkson — APPROVED** [review]

**Review by sbishel — APPROVED** [review]

**Pull request merged**